zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -14,193 +14,240 @@
+        y2s3 = self.data.y2s_3
+        y2s4 = self.data.y2s_4
+        x2 = self.data.x2
         cid = self.data.cid
         close_b = self.data.close_flag
+
         # Convert balances from mutez -> fr
         cust_b = sp.local('cust_b', sp.fst(sp.ediv(custBal, sp.mutez(1)).open_some()))
         cust_bal_b = sp.local("cust_bal_b", sp.mul(cust_b.value, sp.bls12_381_fr("0x01")))
         merch_b = sp.local('merch_b', sp.fst(sp.ediv(merchBal, sp.mutez(1)).open_some()))
         merch_bal_b = sp.local("merch_bal_b", sp.mul(merch_b.value, sp.bls12_381_fr("0x01")))
+
         # Convert the rev_lock from bytes -> fr
         rev_lock_packed = sp.local('rev_lock_packed', sp.concat([sp.bytes("0x050a00000020"), revLock]))
         rev_lock_b = sp.local('rev_lock_b', sp.unpack(rev_lock_packed.value, t = sp.TBls12_381_fr).open_some())
 
         # Verify PS signature against the message
         pk = [y2s0, y2s1, y2s2, y2s3, y2s4]
-        # channel ID, close flag, rev lock, Cust Bal and Merch Bal
+        # the message is composed of the channel ID, close flag, rev lock, cust closing balance and 
+        # merchant closing balance.
         msg = [cid, close_b, rev_lock_b.value, cust_bal_b.value, merch_bal_b.value]
         prod1 = sp.local('prod1', x2)
         for i in range(0, len(msg)):
             prod1.value += sp.mul(pk[i], msg[i])
+        # Compute the pairing check.
         sp.verify(sp.pairing_check([sp.pair(s1, prod1.value), sp.pair(s2, -g2)]), message="pairing check failed")
-        # Update on-chain state and transfer merchant's balance   
+        ## End of PS signature verification
+
+        # Update the closing balances in the contract storage.  
         self.data.custBal = custBal
         self.data.revLock = revLock
+        # Set delayExpiry to the current time plus the specified delay period, selfDelay. When 
+        # delayExpiry has passed, the customer will be able to claim their balance by calling 
+        # custClaim. 
         self.data.delayExpiry = sp.now.add_seconds(self.data.selfDelay)
-        # Pay merchant immediately (unless amount is 0)
-        # Note that all addresses must be implicit accounts (tz1), not smart contracts
+        # Pay merchant immediately (unless amount is 0).
         sp.if merchBal != sp.tez(0):
             sp.send(self.data.merchAddr, merchBal)
-        self.data.merchBal = sp.tez(0)
+        # Set the channel status to CUST_CLOSE.
         self.data.status = CUST_CLOSE
  
-    # merchDispute can be called if the merchant has the secret corresponding
-    # to the latest custClose state. If the secret is valid, the merchant will
-    # receive the customer's balance too.
+    # merchDispute can be called by the merchant if they have the revocation secret corresponding 
+    # to the revocation lock in the customer's custClose entrypoint call. If the secret is valid, 
+    # the merchant will receive the customer's balance.
     @sp.entry_point
     def merchDispute(self, secret):
```

Comment:
can we use the same notation for the revocation secret as is used in zeekoe?

Reply:
making a note at the bottom to update all the terms before merging

Reply:
In zeekoe we call this `revocation_secret`